File: src/app/model/issues/IssueField.ts
import {IssueModelField} from "./IssueModelField";

export class IssueField extends IssueModelField{
  private _value: string

  constructor(title: string, description: string, required: boolean, value: string) {
    super(title, description, required);
    this._value = value;
  }

  get value(): string {
    return this._value;
  }

  set value(value: string) {
    this._value = value;
  }
}

File: src/api/src/main/java/fr/iut/uca/v1/model/issues/IssueField.java
package fr.iut.uca.v1.model.issues;

public class IssueField extends IssueModelField {

    private String value;

    public IssueField(String title, String description, boolean required, String value) {
        super(title, description, required);
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;

        IssueField that = (IssueField) o;

        return value.equals(that.value);
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + value.hashCode();
        return result;
    }
}
